spring-integration-core/src/test/java/org/springframework/integration/endpoint/PseudoTransactionalMessageSourceTests.java

```diff
@@ -342,18 +342,25 @@ public void testRollbackWithManagerUsingStatus() {
 
 	@Test
 	public void testInt2777UnboundResourceAfterTransactionComplete() {
+		MessageSource<?> source = () -> null;
 		SourcePollingChannelAdapter adapter = new SourcePollingChannelAdapter();
+		adapter.setSource(source);
+		adapter.setOutputChannelName("bar");
+		adapter.setBeanFactory(new TestUtils.TestApplicationContext());
```

Comment:
`TestUtils.createTestApplicationContext()` is there for you.
OTOH you always can do `mock(BeanFactory.class)` if that satisfies a requirement.